### ISSUE TITLE
Orientation changes mess up the list

### ISSUE THREAD

**Issue by SudoPlz** [Flipkart/recyclerlistview#145]
Hey there,

here's the issue that I'm having:

<img src="https://cl.ly/1d1v313s3L1t/Screen%20Recording%202018-04-05%20at%2001.20%20PM.gif" />

Note that all works well when I'm on portrait, but once I go in landscape, all hell breaks loose.
As you can see the items don't correctly on landscape, and you can see 2 items instead of one, even though their widths are correct.

I've created a repo with exactly that issue [here](https://github.com/SudoPlz/recyclerlistview-calendar-example/blob/master/src/MiniCalendarManager.js#L880:L907), would you be kind enough to shed some light on this?

Thanks

**Comment by AbdallaMohamed** [comment]
It's a similar problem to preserving scroll position

Maybe you can try https://github.com/Flipkart/recyclerlistview/issues/127#comment-368580060, but instead of applying that logic when getting new data, you apply it when an orientation change happens. 
`Dimensions.addEventListener('change', ...)`

**Comment by SudoPlz** [comment]
Not sure if I got this right, but my problem does not stop once the orientation changes.
Even if I fix the scroll position, if the user tries to scroll, it will still snap to the wrong x value.

**Comment by AbdallaMohamed** [comment]
Have you tried using the canChangeSize prop?

**Comment by SudoPlz** [comment]
Yeah, makes no difference.

**Comment by naqvitalha** [comment]
`canChangeSize` should handle it. For some reason on my machine it isn't running the iOS build. It doesn't resize on Android. Can you run this on expo so that I can have a look?

**Comment by SudoPlz** [comment]
It's using a native module (called `react-native-orientation`), and I haven't linked that on Android, so maybe that's why it's not working on Android.

It should work on iOS, what error do you get? Maybe I can fix it.

Also, I can create an expo, but does expo support native modules (I thought it didn't)?

I'll do anything to sort this out.

**Comment by naqvitalha** [comment]
Why do you need a native module to detect orientation change? Check #58

**Comment by SudoPlz** [comment]
Ok, I wasn't aware of that, but never the less, can you please tell me the issue iOS is giving you?
or should we skip that and create an expo project from scratch?

I really need to focus on the real issue here :/

**Comment by naqvitalha** [comment]
If you can reproduce this on Android I will take a look ASAP. I think I need to upgrade Mac OS and XCode to run the project. That's why I always use expo for iOS. Expo or, Android repro will do.

**Comment by SudoPlz** [comment]
Perfect, @naqvitalha the Android project should now be working on that same repo.

Try it and let me know how it went.

**Comment by naqvitalha** [comment]
@SudoPlz Dude I tried on Android post your changes and saw the same issue. I added `canChangeSize` and it works perfectly post that. `canChangeSize` was designed for such scenarios. Try that maybe you did something wrong before.

**Comment by SudoPlz** [comment]
Whaaat? But I did add `canChangeSize` and tested on iOS, and I got the same issue. :S
I'll try on Android as well!!

**Comment by naqvitalha** [comment]
Yeah :P try and let me know.

**Comment by SudoPlz** [comment]
Pffff.... it does work on the reproduction repo, but not on my main project. I might be using a slightly older version though. I'll try to update and see if that changes.

**Comment by SudoPlz** [comment]
Ok, so I confirm that after I updated to the latest version `canChangeSize` did work, thanks a ton for your time @naqvitalha and sorry for wasting your time.

You see I'm not using your library through npm, I'm using a build locally, because I wanted to be able to modify and debug your library in real time, without having to care about typescript.

The reason is https://github.com/Flipkart/recyclerlistview/issues/103 

Closing that issue for now.
